### ISSUE TITLE
Namespaced model not working after STI fix added

### ISSUE THREAD

**Issue by ggordon** [JSONAPI-Resources/jsonapi-resources#523]
If my path for the model and resource differ, the resource cannot be found. 

```
module Legacy
  class FlatPost < ActiveRecord::Base
    self.table_name = "posts"
  end
end

class FlatPostResource < JSONAPI::Resource
  model_name "::Legacy::FlatPost"
  attribute :title
end

class FlatPostsController < JSONAPI::ResourceController
end 
```

and a route

```
jsonapi_resources :flat_posts
```

then 

```
get '/flat_posts' 
```

Gives me
**Could not find resource 'legacy/flat_post'.**

Where did that `legacy` come from?

```
Started GET "/flat_posts" for 127.0.0.1 at 2015-11-02 14:51:52 -0500
Processing by FlatPostsController#index as HTML
  Legacy::FlatPost Load (0.3ms)  SELECT "posts".* FROM "posts"  ORDER BY "posts"."id" ASC
Internal Server Error: JSONAPI: Could not find resource 'legacy/flat_post'. (Class Legacy::FlatPostResource not found) /Code/jsonapi-resources/lib/jsonapi/resource.rb:300:in `resource_for'
//Code/jsonapi-resources/lib/jsonapi/resource.rb:525:in `block in find
```

**Comment by ggordon** [comment]
It looks like `JSONAPI::Resource#resource_type_for(model)` is confusing model path with resource path, most of the time these are the same, but not always. I'm at a loss as to how to fix this, any ideas @lgebhardt ?

**Comment by lgebhardt** [comment]
@ggordon Would it help to make it a requirement than models and resources must reside in the same namespace? It's been something I've been thinking about since seeing #513. This would be a breaking change, but I think it might be for the best. But I haven't done any testing of the idea so I don't know exactly what will break if we go that route.

**Comment by ggordon** [comment]
@lgebhardt Unfortunately that won't work for me. I have some legacy namespaced models and the resource that wraps it is in a different namespace. This was all working before. I think I might need to create a mapping like `@@resource_types`, but mapping models to resources, or maybe just overrides when there are differences. (How did I miss #513 it's exactly the same issue?)

**Comment by lgebhardt** [comment]
Would it be possible to create a model in the resource's namespace that is derived from the legacy model?

**Comment by barelyknown** [comment]
I _always_ namespace the API endpoints (`v1`, `v2`, ...). It'd be a pretty major pain to have to derive a matching model for every resource in the same namespace. A big :-1: from me on that change.

**Comment by bajalovic** [comment]
I have a model at `app/models/weather/alert.rb`, and I created a resource at `app/resources/api/v1/weather_alert_resource.rb`.

I extended the `resource_type_for(model)` in my `WeatherAlertResource` like this:

``` ruby
    class Api::V1::WeatherAlertResource < JSONAPI::Resource
      attributes :country, :town, :from, :expires, :description, :alert_type
      model_name 'Weather::Alert'

      class << self
        def resource_type_for(model)
          'api/v1/weather_alert'
        end
      end
    end
```

and now I can make a `GET /api/v1/weather_alerts` and everything works fine.

**Comment by ggordon** [comment]
@bajalovic That's a nice work around especially since (for me at least) there's only a few model/resource name mismatches. 
Doesn't seem to be sufficient when this is a relationship on another resource.

**Comment by ggordon** [comment]
Fixed by #528
